contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -459,9 +467,12 @@ func (s *SchedulerServer) prepareExecutorInfo(hks hyperkube.Interface) (*mesos.E
 
 	if s.kubeletKubeconfig != "" {
 		//TODO(jdef) should probably support non-local files, e.g. hdfs:///some/config/file
-		uri, basename := s.serveFrameworkArtifact(s.kubeletKubeconfig)
-		ci.Uris = append(ci.Uris, &mesos.CommandInfo_URI{Value: proto.String(uri)})
-		ci.Arguments = append(ci.Arguments, fmt.Sprintf("--kubeconfig=%s", basename))
+		if s.kubeletKubeconfig != s.proxyKubeconfig {
+			// allows kubelet-kubeconfig and proxy-kubeconfig to point to the same file
```

Comment:
maybe a comment here along the lines of: if kubelet-kubeconfig and proxy-kubeconfig point to different files with the same basename this will fail immediately. that's expected given that we serve files at endpoints like `/$(basename $file)`. or even better, perhaps a panic handler here that generates a user-friendly message to check for this condition upon a panic? we don't have such in other places but i'm a little concerned that users could end up in this situation if they, for example, have all of their config files named the same but split into different directories: `/myconfig/{scheduler,kubelet,proxy}/kubeconfig`

thoughts?